### PR TITLE
Support json type

### PR DESCRIPTION
**Pull request by Tigrov**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: src/Builder/JsonExpressionBuilder.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * Builds expressions for {@see `Yiisoft\Db\Expression\JsonExpression`} for SQLite.
+ */
+final class JsonExpressionBuilder implements ExpressionBuilderInterface
+{
+    public function __construct(private QueryBuilderInterface $queryBuilder)
+    {
+    }
+
+    /**
+     * The Method builds the raw SQL from the $expression that won't be additionally escaped or quoted.
```
Comment (vjik) [inline src/Builder/JsonExpressionBuilder.php:29]:
```suggestion
     * The method builds the raw SQL from the `$expression` that won't be additionally escaped or quoted.
```

File: src/ColumnSchema.php [hunk 2]
```diff
@@ -32,4 +38,48 @@
...
+     * @return mixed The converted value.
+     */
+    public function phpTypecast(mixed $value): mixed
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if ($this->getType() === SchemaInterface::TYPE_JSON) {
+            return json_decode((string) $value, true, 512, JSON_THROW_ON_ERROR);
```
Comment (vjik) [inline src/ColumnSchema.php:80]:
```suggestion
            return Json::decode((string) $value);
```

**Review by vjik — COMMENTED** [review]
Overall looks good 👍 Please, add line to changelog

File: src/ColumnSchema.php (hunk 2 above)
Comment (vjik) [reply src/ColumnSchema.php:80]: Or you can replace `Json::encode()` in other place to `json_encode` and drop dependency `yiisoft/json`. It's even better this way.

Comment (Tigrov) [reply src/ColumnSchema.php:80]: `Json::encode()` is used according to MySQL and PosgreSQL

https://github.com/yiisoft/db-mysql/blob/master/src/Builder/JsonExpressionBuilder.php#L53
https://github.com/yiisoft/db-pgsql/blob/master/src/Builder/JsonExpressionBuilder.php#L56

`Json::encode()` Pre-processes the data before sending it to `json_encode()`:
https://github.com/yiisoft/json/blob/master/src/Json.php#L100

`Json::decode()` is the same as `json_decode()` but with default arguments:
https://github.com/yiisoft/json/blob/master/src/Json.php#L81

**Review by vjik — APPROVED** [review]

File: src/ColumnSchema.php (hunk 2 above)
Comment (xepozz) [reply src/ColumnSchema.php:80]: Still good idea to use the helper's function

Comment (Tigrov) [reply src/ColumnSchema.php:80]: @xepozz Do you mean use `Json::decode()`?

MySQL and PostgreSQL use `json_decode()` directly
https://github.com/yiisoft/db-mysql/blob/master/src/ColumnSchema.php#L59
https://github.com/yiisoft/db-pgsql/blob/master/src/ColumnSchema.php#L145

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**